app/javascript/components/project-submissions/components/__tests__/submission.test.jsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+
+import ProjectSubmissionContext from '../../ProjectSubmissionContext';
+import Submission from '../submission';
+
+const defaultSubmission = {
+  user_name: 'Example User',
+  repo_url: 'http://example.com/repo',
+  live_preview_url: 'https://google.com',
+  is_public: true,
+  lesson_has_live_preview: true,
+  user_id: 1,
+};
+
+const renderSubmissionComponent = (submission, userId = 1) => render(
+  <ProjectSubmissionContext.Provider value={{
+    userId,
+  }}
+  >
+    <Submission
+      submission={submission}
+      handleUpdate={jest.fn()}
+      onFlag={jest.fn()}
+      handleDelete={jest.fn()}
+      handleLikeToggle={jest.fn()}
+    />
+    ,
+  </ProjectSubmissionContext.Provider>,
+);
+
+describe('Submission: Current User', () => {
+  let queryByTestId;
+
+  beforeEach(() => {
+    ({ queryByTestId } = renderSubmissionComponent(defaultSubmission));
+  });
+
+  afterAll(() => {
+    jest.resetModules();
+  });
+
+  test('Renders edit button', () => {
+    const editButton = queryByTestId('edit-submission-btn');
+    expect(editButton).toBeInTheDocument();
+  });
+
+  test('Displays edit modal when edit button is clicked.', () => {
+    fireEvent.mouseDown(queryByTestId('edit-submission-btn'));
+
+    const editForm = queryByTestId('edit-form');
+    expect(editForm).toBeInTheDocument();
+  });
+
+  test("Doesn't render report button", () => {
+    const reportButton = queryByTestId('flag-btn');
+    expect(reportButton).not.toBeInTheDocument();
+  });
+});
+
+describe('Submission: Different User', () => {
+  let queryByTestId;
+
+  beforeEach(() => {
+    ({ queryByTestId } = renderSubmissionComponent(defaultSubmission, 2));
+  });
+
+  afterAll(() => {
+    jest.resetModules();
+  });
+
+  test('Renders edit button', () => {
```

Comment:
"Does not render edit button"

Reply:
Will work on this soon, thanks for the feedback.

Reply:
Done. Here's what the final tests look like:
![image](https://user-images.githubusercontent.com/12211990/130318455-33b7ab3e-7611-4203-9e34-231ff8b971ff.png)